### PR TITLE
Added Andes badge

### PR DESCRIPTION
**Pull request by maaguero**
## Useful links
- [Contributing](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CONTRIBUTING.md) has more information and tips for a great pull request. This will give you a nice introduction about how to contribute with Andes.
- [Wiki](https://github.com/mercadolibre/fury_andesui-android/wiki) has all the info you need to start with Andes. Even I heard there's a video there...

## Motivation
Implement the AndesBadge (Pill) component defined by the UX team

## Description
AndesBadge (Pill) is a component defined here.
- How does it work?

## Affected component
AndesBadge

## Frontify component link
[AndesBadge Frontify](https://company-161429.frontify.com/d/kxHCRixezmfK/n-a#/componentes/badge)

## Screenshots

image
image

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [x] I have coded an example inside the Demo App,
- [x] I have added proper tests,
- [ ] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [ ] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [x] This code adds a new component
- [ ] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/src/main/res/badge/layout/andes_layout_badge.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    tools:parentTag="android.support.v7.widget.CardView">
+
+    <TextView
+        android:id="@+id/andes_badge_text"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+
```
Comment (lautimfernandez) [inline components/src/main/res/badge/layout/andes_layout_badge.xml:14]: empty line

File: components/src/main/res/badge/layout/andes_layout_badge.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        android:layout_marginStart="6dp"
+
```
Comment (lautimfernandez) [inline components/src/main/res/badge/layout/andes_layout_badge.xml:16]: empty line

File: components/src/main/java/com/mercadolibre/android/andesui/badge/border/AndesBadgeBorderInterface.kt [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+ *
+ */
+internal class AndesCornerBadgeBorder : AndesBadgeBorderInterface {
+
+    override fun upStartCornerRadius(size: AndesBadgeSizeInterface, context: Context) = NO_ROUND_RADIUS
+    override fun upEndCornerRadius(size: AndesBadgeSizeInterface, context: Context) = size.cornerRadius(context)
+    override fun bottomStartCornerRadius(size: AndesBadgeSizeInterface, context: Context) = size.cornerRadius(context)
+    override fun bottomEndCornerRadius(size: AndesBadgeSizeInterface, context: Context) = NO_ROUND_RADIUS
+
+    companion object {
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/border/AndesBadgeBorderInterface.kt:64]: para evitar hacer esto podriamos hacer que la clase padre siempre retorne la constante NO_ROUND_RADIUIS = 0 y solamente hacer el override en las que se necesita del size.

File: components/src/main/java/com/mercadolibre/android/andesui/badge/size/AndesBadgeSizeInterface.kt [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ */
+internal interface AndesBadgeSizeInterface {
+    /**
+     * Returns a [Float] representing the text size to be used.
+     *
+     * @param context needed for accessing some resources.
+     * @return a [Float] representing the text size to be used.
+     */
+    fun textSize(context: Context): Float
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/size/AndesBadgeSizeInterface.kt:19]: dado que todas hacen override podrian ser abstractas

File: components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadge.kt [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+    private fun createConfig() = AndesBadgeConfigurationFactory.create(context, andesBadgeAttrs)
+
+    companion object {
+        private const val CARD_ELEVATION = 0F
+        private val BORDER_DEFAULT = AndesBadgeBorder.ROUNDED
+        private val HIERARCHY_DEFAULT = AndesBadgeHierarchy.LOUD
+        private val MODIFIER_DEFAULT = AndesBadgeModifier.PILL
+        private val SIZE_DEFAULT = AndesBadgeSize.SMALL
+        private val STATE_DEFAULT = AndesBadgeType.NEUTRAL
+        private val TITLE_DEFAULT = null
```
Comment (lautimfernandez) [inline components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadge.kt:216]: ponerle text como la variable

File: components/src/main/java/com/mercadolibre/android/andesui/badge/size/AndesBadgeSizeInterface.kt (hunk 4 above)
Comment (maaguero) [reply components/src/main/java/com/mercadolibre/android/andesui/badge/size/AndesBadgeSizeInterface.kt:19]: Al ser una interfaz ya son abstractas.
Voy a convertir la AndesBadgeTypeInterface a interfaz para que quede consistente.

Comment (lautimfernandez) [reply components/src/main/java/com/mercadolibre/android/andesui/badge/size/AndesBadgeSizeInterface.kt:19]: es verdad, malisima mia. Excelente igual.

File: components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadge.kt [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+    }
+
+    /**
+     * Creates all the views that are part of this badge.
+     * After a view is created then a view id is added to it.
+     *
+     */
+    private fun initComponents() {
+        val container = LayoutInflater.from(context).inflate(R.layout.andes_layout_badge, this)
+        //badgeBackground = container.findViewById(R.id.andes_badge_background)
```
[the remark this answers is not in the document]
Comment (maaguero) [reply components/src/main/java/com/mercadolibre/android/andesui/badge/AndesBadge.kt:157]: No es necesaria esa linea, ni esa variable.

**Pull request merged**